test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java

```diff
@@ -740,12 +742,14 @@ public void close() {
                     if (oldState.nodeEnvironment != null) {
                         nodeEnvironment = oldState.nodeEnvironment;
                         final MetaData updatedMetaData = adaptGlobalMetaData.apply(oldState.getLastAcceptedState().metaData());
-                        if (updatedMetaData != oldState.getLastAcceptedState().metaData()) {
-                            throw new AssertionError("TODO adapting persistent metadata is not supported yet");
-                        }
                         final long updatedTerm = adaptCurrentTerm.apply(oldState.getCurrentTerm());
-                        if (updatedTerm != oldState.getCurrentTerm()) {
-                            throw new AssertionError("TODO adapting persistent current term is not supported yet");
+                        if (updatedMetaData != oldState.getLastAcceptedState().metaData() || updatedTerm != oldState.getCurrentTerm()) {
```

Comment:
while not strictly part of the logic in this PR, this fixes the `testCannotJoinClusterWithDifferentUUID` test that was occasionally failing without this being properly implemented.

Reply:
👍